etna/models/nn/deepar_new.py

```diff
@@ -0,0 +1,383 @@
+from collections import Counter
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import Optional
+from typing import Type
+
+import numpy as np
+import pandas as pd
+from typing_extensions import TypedDict
+
+from etna import SETTINGS
+from etna.distributions import BaseDistribution
+from etna.distributions import FloatDistribution
+from etna.distributions import IntDistribution
+from etna.models.base import DeepBaseModel
+from etna.models.base import DeepBaseNet
+
+if SETTINGS.torch_required:
+    import torch
+    import torch.nn as nn
+    from torch.distributions import NegativeBinomial
+    from torch.distributions import Normal
+    from torch.utils.data.sampler import Sampler
+
+
+class DeepARSampler(Sampler):
+    """Select samples with probabilities 1 / number of appearance of given segment."""
+
+    def __init__(self, data):
+        self.data = data
+
+    def __iter__(self):
+        segments = [d["segment"] for d in self.data]
+        count_segments = Counter(segments)
+        p = torch.tensor([1 / count_segments[segment] for segment in segments])
+        num_samples = len(self.data) // len(set(segments))  # TODO is good?
+        idx = torch.multinomial(p, num_samples=num_samples)
+        return iter(idx)
+
+    def __len__(self):
+        return len(self.data)
+
+
+class DeepARBatchNew(TypedDict):
+    """Batch specification for DeepAR."""
+
+    encoder_real: "torch.Tensor"
+    decoder_real: "torch.Tensor"
+    encoder_target: "torch.Tensor"
+    decoder_target: "torch.Tensor"
+    segment: "torch.Tensor"
+    weight: "torch.Tensor"
+
+
+class DeepARNetNew(DeepBaseNet):
+    """DeepAR based Lightning module with LSTM cell."""
+
+    def __init__(
+        self,
+        input_size: int,
+        num_layers: int,
+        hidden_size: int,
+        lr: float,
+        loss: Type[DeepARSampler],
+        optimizer_params: Optional[dict],
+    ) -> None:
+        """Init DeepAR.
+
+        Parameters
```

Comment:
Add several parameters
1. `scale: bool` -- wether to do the scaling with weights, if False weights = 1
2. `n_samples: Optional[int] = None`
- None -- use theoretical mean of distribution as predictions
- int -- do inference `n_samples` times using samples from distribution as inputs to the next steps, use imperical mean of samples as predictions